hack/community/generate.sh

```diff
@@ -0,0 +1,133 @@
+    " "${CO_CSV}"
+
+    # Replace annotations fields
+    # TODO: use yq for the annotations.yaml replacement
+    sed -i -e "s/"preview,stable"/$MATURITY/" -e "s/"stable"/$MATURITY/" "${CO_ANNOTATIONS}"
+}
+
+# copy the WMCO bundle and its contents to a new community folder. Rename files
+# as needed.
+generate_manifests() {
+  local CO_DIR=$1
+  local OPERATOR_VERSION=$2
+  local OPERATOR_MANIFESTS=$3
+  local OPERATOR_METADATA=$4
+  local DESCRIPTION=$5
+  local QUAY_IMAGE=$6
+  local CO_OPERATOR_DIR="${CO_DIR}/operators/community-windows-machine-config-operator"
+
+  echo "Update operator manifests"
+  mkdir "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}"
+  cp -r "${OPERATOR_MANIFESTS}" "${OPERATOR_METADATA}" "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}"
+  mv "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/windows-machine-config-operator.clusterserviceversion.yaml" "${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/community-windows-machine-config-operator.${OPERATOR_VERSION}.clusterserviceversion.yaml"
+  pushd "${CO_DIR}"
+  git checkout -B "${OPERATOR_VERSION}"
+  CO_CSV="${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/manifests/community-windows-machine-config-operator.${OPERATOR_VERSION}.clusterserviceversion.yaml"
+  CO_ANNOTATIONS="${CO_OPERATOR_DIR}/${OPERATOR_VERSION}/metadata/annotations.yaml"
+
+  replace "$CO_CSV" "$OPERATOR_VERSION" "$DESCRIPTION" "$QUAY_IMAGE" "$CO_ANNOTATIONS"
+}
+
+# Create a signed commit for the community-operators-prod repo
+commit() {
+  local OPERATOR_VERSION=$1
+  TITLE="operators community-windows-machine-config-operator.v$OPERATOR_VERSION"
+
+  echo "commit changes"
+  git add --all
+  git commit -s -m "${TITLE}"
+  popd
+}
+
+OPERATOR_VERSION="$1"
+COMMUNITY_VERSION="$2"
+QUAY_IMAGE=REPLACE_IMAGE
+OPERATOR_MANIFESTS="bundle/manifests"
+OPERATOR_METADATA="bundle/metadata"
+CO_PROJECT="community-operators-prod"
+CO_REPO="redhat-openshift-ecosystem/${CO_PROJECT}"
+CO_FORK="https://${GITHUB_USER}:${GITHUB_TOKEN}@github.com/${GITHUB_USER}/${CO_PROJECT}.git"
+
+# Check if yq is installed
+if ! command -v which yq &> /dev/null; then
+  echo "Please install yq before running generate.sh. Directions to install yq are in the readme.md"
+  exit 1
+fi
+
+# check if ENV vars are set
+if [ -z "${GITHUB_USER}" ] || [ -z "${GITHUB_TOKEN}" ] || [ -z "${GITHUB_NAME}" ]; then
+    echo "Must set all of GITHUB_USER, GITHUB_TOKEN, and GITHUB_NAME"
+    exit 1
+fi
+
+# check if QUAY_IMAGE is replaced
+if [ "$QUAY_IMAGE" == "REPLACE_IMAGE" ]; then
+    echo "Must set QUAY_IMAGE to latest community tag"
+    exit 1
+fi
+
+echo "Cutting community release for v${OPERATOR_VERSION}"
+
+# Inject appropriate community-version into the description
+DESCRIPTION=$(cat hack/community/csv/description.md)
+DESCRIPTION=${DESCRIPTION//COMMUNITY_VERSION/$COMMUNITY_VERSION}
+
+# Check if user has already cloned community-operators-prod
+if [ -z $CO_DIR ]; then
+   CO_DIR=$(mktemp -d)
+   clone_community "$CO_FORK" "$CO_DIR" "$CO_REPO"
```

Comment:
I'm now thinking this is doing a little too much, and that CO_DIR should be made a required env var.
The user is going have to keep their fork maintained anyway to use this script, they should already have it cloned.

i'd prefer the cloning be removed from this script completely.

Reply:
That would mean that the env var's required for cloning should be removed from the requirements in the README